Fix Extended RTTI static build

MaartenBent opened the pull request:
Fix declaration of 'o' hides previous local declaration warnings.
Change two wxCONSTRUCTOR_5 definitions using wxBitmapBundle instead of wxBitmap (this does not give build errors).
wxGenericCalendarCtrl is missing all XTI implementations, so just use RTTI.
And fix building the xti sample.

Note the shared build still fails due to WXDLLIMPEXP related issues.

Closes #22300

vadz commented on samples/xti/xti.cpp:
```diff
@@ -277,13 +277,13 @@ class MyDesignerPersister : public wxObjectWriterCallback
         // be connected really in the designer, so we have to supply
         // the information
-        const wxObject* but = wxAnyGetAsObjectPtr( m_frame->GetProperty("Button") );
+        const wxObject* but = wxAnyGetAsObjectPtr( m_frame->GetProperty(wxString("Button")) );
```
Shouldn't this be
```suggestion
        const wxObject* but = wxAnyGetAsObjectPtr( m_frame->GetProperty(wxT("Button")) );
```
?

MaartenBent replied:
`wxT` works too, I'll update it.

I don't feel like changing all the xti code to accept `wxString` instead of `wxChar` (it's not only properties). Especially since no one enables `wxUSE_EXTENDED_RTTI`.

vadz commented on include/wx/object.h:
```diff
@@ -372,6 +372,10 @@ class wxObjectDataPtr
 
 class WXDLLIMPEXP_BASE wxObject
 {
+#if wxUSE_EXTENDED_RTTI
+    wxDECLARE_DYNAMIC_CLASS(wxObject);
+#endif
```
Sorry but why do we need it here now?

MaartenBent replied:
You removed it in 95c98a0b5ff71caca6654327bf341719c6587766 0585c96a3bceae440cfe455687e62def82ed8d75 and manually defined `ms_classInfo`.

While RTTI only has `ms_classInfo`, XTI has a lot more static members that now all need to be defined in `object.h`. The simpler solution is to keep using `wxDECLARE_DYNAMIC_CLASS`.

https://github.com/wxWidgets/wxWidgets/blob/ce46d773609154f9e6c663e08aca96e2314db23d/include/wx/xti.h#L437-L452

vadz replied:
I see, thanks, I should have guessed I must have been broken it myself...

vadz reviewed (commented):
Thanks for fixing this, but I'm curious about the changes to `wxObject`. They don't change anything for the default build, of course, but, still, it seems strange to change something as basic as this now, after so many years, so why is it needed?

MaartenBent committed:
Fix Extended RTTI static build

Fix declaration of 'o' hides previous local declaration warnings.
Change two wxCONSTRUCTOR_5 definitions using wxBitmapBundle instead of wxBitmap (this does not give build errors).
wxGenericCalendarCtrl is missing all XTI implementations, so just use RTTI.
And fix building the xti sample.

Note the shared build still fails due to WXDLLIMPEXP related issues.

Closes #22300
